packages/cdk-assets/lib/aws.ts

```diff
@@ -228,6 +235,8 @@ export class DefaultAwsClient implements IAws {
       config.region = options.region;
       if (options.assumeRoleArn) {
         config.credentials = fromTemporaryCredentials({
+          // dont forget the credentials chain.
+          masterCredentials: this.mainCredentials,
```

Comment:
We have something similar in CDK CLI:

https://github.com/aws/aws-cdk-cli/blob/1bef26344b3e311fbac7241ffa9a3e05b195d6de/packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts#L357